### PR TITLE
iOS PlatformView clip path

### PR DESCRIPTION
**Pull request by cyanglaz**
Add clip path to the iOS platform view.

Quarts doesn't support Conic path. We will use a simple quad path to approximate in this PR. An issue is filed for tracking the effort for that https://github.com/flutter/flutter/issues/35062
Also, https://github.com/flutter/flutter/pull/35063 is created in the framework to document on the UIKitView to explain this as well.

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm [hunk 1]
```diff
@@ -260,14 +260,6 @@
...
     ++iter;
   }
-
-  // Reverse scale based on screen scale.
-  //
-  // The UIKit frame is set based on the logical resolution instead of physical.
-  // (https://developer.apple.com/library/archive/documentation/DeviceInformation/Reference/iOSDeviceCompatibility/Displays/Displays.html).
-  // However, flow is based on the physical resolution. For eaxmple, 1000 pixels in flow equals
-  // 500 points in UIKit. And until this point, we did all the calculation based on the flow
-  // resolution. So we need to scale down to match UIKit's logical resolution.
```
Comment (amirh) [inline shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm]: Is this comment no longer true?

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.mm [hunk 2]
```diff
@@ -129,6 +131,64 @@ - (void)clipRRect:(const SkRRect&)clipSkRRect {
...
+  SkPoint pts[kMaxPointsInVerb];
+  SkPath::Verb verb = iter.next(pts);
+  while (verb != SkPath::kDone_Verb) {
+    switch (verb) {
+      case SkPath::kMove_Verb: {
+        CGPathMoveToPoint(pathRef, nil, pts[0].x(), pts[0].y());
+        break;
+      }
+      case SkPath::kLine_Verb: {
+        CGPathAddLineToPoint(pathRef, nil, pts[1].x(), pts[1].y());
```
Comment (amirh) [inline shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.mm]: are we sure we can assume that pts[0] is the last point of the previous verb?

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm (hunk 1 above)
Comment (cyanglaz) [reply shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm]: Removed by accident, adding it back.

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.mm (hunk 2 above)
Comment (cyanglaz) [reply shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.mm]: Offline discussion: We confirmed we can assume the pts[0] is always the last point of the previous verb, but we are going to add a DCHECK for our own sanity.

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.mm [hunk 3]
```diff
@@ -129,6 +131,74 @@ - (void)clipRRect:(const SkRRect&)clipSkRRect {
...
+        last_pt_from_last_verb = pts[1];
+        break;
+      }
+      case SkPath::kQuad_Verb: {
+        CGPathAddQuadCurveToPoint(pathRef, nil, pts[1].x(), pts[1].y(), pts[2].x(), pts[2].y());
+        last_pt_from_last_verb = pts[2];
+        break;
+      }
+      case SkPath::kConic_Verb: {
+        // Conic is not available in quartz, we use quad to approximate.
```
Comment (amirh) [inline shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.mm]: Reference a TODO issue here to better approximate this with multiple quads.

File: flow/embedded_views.h [hunk 4]
```diff
@@ -120,7 +120,6 @@ class MutatorsStack {
   void pushClipRect(const SkRect& rect);
   void pushClipRRect(const SkRRect& rrect);
   void pushClipPath(const SkPath& path);
-
```
Comment (amirh) [inline flow/embedded_views.h]: revert

**Pull request merged**